crates/core/src/evm/mod.rs

```diff
@@ -43,8 +45,9 @@ pub fn execute_tx(
         .with_external_context(TracerEip3155::new(Box::new(std::io::stderr())).without_summary())
         .append_handler_register(inspector_handle_register)
         .build();
-    let tx_result = evm.transact().unwrap();
-    tx_result.result.into()
+    evm.preverify_transaction().map_err(EvmError::from)?;
+    let tx_result = evm.transact().map_err(EvmError::from)?;
```

Comment:
[`transact` already verifies the transaction](https://docs.rs/revm/latest/revm/struct.Evm.html#method.transact):
> This function will validate the transaction

Reply:
Oh